src/rpcrequest-payload.ts

```diff
@@ -106,9 +126,76 @@ class RpcRequestPayload implements Iterable<Buffer> {
       param.collation = this.collation;
     }
 
-    yield type.generateTypeInfo(param, this.options);
-    yield type.generateParameterLength(param, this.options);
-    yield * type.generateParameterData(param, this.options);
+    const typeINfo = type.generateTypeInfo(param, this.options);
+    yield typeINfo;
+    if (writeValue) {
+      yield type.generateParameterLength(param, this.options);
+      yield * type.generateParameterData(param, this.options);
+    }
+  }
+
+  *_generateEncryptedParameter(parameter: Parameter) {
+    const encryptedParam = {
+      value: parameter.encryptedVal,
+      type: VarBinary,
+      output: parameter.output,
+      name: parameter.name,
+      forceEncrypt: parameter.forceEncrypt
+    };
+    yield* this.generateParameterData(encryptedParam, true);
+    yield* this.generateParameterData(parameter, false);
+    yield* this._writeEncryptionMetadata(parameter.cryptoMetadata);
+
+  }
+
+  _encryptParameters(parameters: Parameter[]): Promise<Parameter[]> | Parameter[] {
+    return new Promise((resolve) => {
+      if (this.options.serverSupportsColumnEncryption === true) {
+        const promises: Promise<void>[] = [];
+
+        for (let i = 0, len = parameters.length; i < len; i++) {
+          const type = parameters[i].type;
+          const paramValue = parameters[i].value;
+          if (parameters[i].cryptoMetadata && (paramValue !== undefined && paramValue !== null)) {
+            if (!type.toBuffer) {
+              throw new Error(`Column encryption error. Cannot convert type ${type.name} to buffer.`);
+            }
+            const plainTextBuffer = type.toBuffer(parameters[i], this.options);
+            if (plainTextBuffer) {
```

Comment:
`plainTextBuffer` can never be `null` here. It will always be a `Buffer` instance.